src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs

```diff
@@ -229,18 +229,17 @@ private bool AllowClientConnection(TcpClient tcpClient)
                 return true;
             }
 
-            var clientRemoteEndPoint = tcpClient.Client.RemoteEndPoint as IPEndPoint;
+            var clientLocalEndPoint = tcpClient.Client.LocalEndPoint as IPEndPoint;
```

Comment:
This is a big change it seems, what the logic behind using `LocalEndPoint`?

Reply:
As per the bitcoin specification on which I believe we are basing our `-whitebind` flag:
```Bind to given address and whitelist peers connecting to it. Use [host]:port notation for IPv6```

Since the option defines interface onto which the node is binding I had to correct the logic in the `AllowClientConnection` to assert if the connection should be "whitelisted" based on the local interface (we do not whitelist remote interfaces as this wasn't the intention of the flag).

Reply:
Can you point to the code in core where this happens?